trl/agents/utils.py

```diff
@@ -0,0 +1,322 @@
+
+    # Apply the tokenizer's chat template
+    processed_prompts = tokenizer.apply_chat_template(conversations, tokenize=False, add_generation_prompt=True)
+
+    # Create a new dataset with processed prompts
+    return dataset.map(lambda x, idx: {prompt_column: processed_prompts[idx]}, with_indices=True)
+
+
+def prepare_data_for_local_agent(
+    dataset,
+    tokenizer,
+    prompt_column: str = "prompt",
+    system_prompt: str = default_system_prompt,
+    tools: List[Callable] = None,
+    include_source_code: bool = True,
+) -> list:
+    """
+    Prepares the Hugging Face dataset for the local agent by constructing conversations
+    and applying the system prompt.
+
+    Args:
+        dataset (`Dataset`):
+            A Hugging Face dataset object
+        tokenizer (`PreTrainedTokenizer`):
+            A tokenizer with an `apply_chat_template` method to process conversations
+        prompt_column (`str`, *optional*, defaults to `"prompt"`):
+            Name of the column containing prompts.
+        system_prompt (`str`, *optional*, defaults to `default_system_prompt`):
+            The base system prompt.
+        tools (`List[Callable]` or `None`, *optional*, defaults to `None`):
+            List of callable functions to include in system prompt
+        include_source_code (`bool`, *optional*, defaults to `True`):
+            If True, includes source code of tools, else includes docstrings
+
+    Returns:
+        `Dataset`:
+            Modified dataset with processed prompts in the prompt column
+    """
+    # Convert dataset prompts to conversational format
+    conversations = [[{"role": "user", "content": prompt}] for prompt in dataset[prompt_column]]
+
+    # Process tools if provided
+    if tools:
+        tools_documentation = "\nAvailable tools:\n\n"
+        for tool in tools:
+            tools_documentation += f"Tool: {tool.__name__}\n"
+            if include_source_code:
+                try:
+                    tools_documentation += f"Source code:\n{getsource(tool)}\n\n"
+                except Exception as e:
+                    tools_documentation += f"Error getting source: {str(e)}\n\n"
+            else:
+                doc = tool.__doc__ or "No documentation available"
+                tools_documentation += f"Documentation:\n{doc}\n\n"
+
+        system_prompt += tools_documentation
+
+    # Create the system prompt message
+    system_message = {"role": "system", "content": system_prompt}
+
+    # Prepend the system message to each conversation
+    for convo in conversations:
+        if not convo or convo[0].get("role") != "system":
+            convo.insert(0, system_message)
+
+    # Apply the tokenizer's chat template
+    processed_prompts = tokenizer.apply_chat_template(conversations, tokenize=False, add_generation_prompt=True)
+
+    # Create a new dataset with processed prompts
+    return dataset.map(lambda x, idx: {prompt_column: processed_prompts[idx]}, with_indices=True)
```

Comment:
Will this remove all of the columns? If so this might break things for folks who put the answer in a column to verify the completion in the reward func.